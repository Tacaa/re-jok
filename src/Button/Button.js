// @flow
import * as React from 'react'

import Icon from '../Icon';

import {
  Btn,
  OutlineBtn,
  Label,
  SpinnerButton,
  IconBtn
} from './style';

import { isChild } from '../helpers/typeUtils';

type Props = {
  /** Set button unable to click or not **/
  disabled?: boolean,
  /** Event when user clicked to button **/
  onClick?: Function,
  /** Set button to a link**/
  hrefString?: string,
  /** The size of button . The default size of button is default **/
  size: 'small' | 'default' | 'medium' | 'large',
  /** Set shape of button **/
  shape?: 'circle' | 'rounded',
  /** Button can show a loading**/
  loading?: boolean,
  /** Set icon in button**/
  icon?: string,
  /** Add more style to button **/
  style?: Object,
  /** Add more class to button **/
  className?: string,
  /** Primary color of button **/
  primary?: boolean,
  /** Secondary color of button **/
  secondary?: boolean,
  /** Set width of button to 100%**/
  fullWidth?: boolean,
  /** move icon to left or right label. The default is left **/
  iconPosition: 'left' | 'right',
  /** if you don't want use primary or secondary , you can set your color to button **/
  backgroundColor?: string,
  /** Make your button animated**/
  animated?: string,
  /** Change tag name of button ... span, div, a v..v **/
  tag?: string,
  /** Change type of button**/
  variant: 'outline' | 'inverted' | 'raised' | 'icon',
  /** create button with box shadow or not**/
  hasBoxShadow: boolean,
  /** Set type of button **/
  rule: 'button' | 'reset' | 'submit',
  /**Children element can be string, number, HTML node anything you want**/
  children?: any,
}

const defaultProps = {
  disabled: false,
  size: 'default',
  rule: 'button',
  loading: false,
  primary: false,
  secondary: false,
  fullWidth: false,
  iconPosition: 'left',
  backgroundColor: "",
  tag: "button",
  variant: 'raised',
  hrefString: "",
  hasBoxShadow: false
};

class Button extends React.Component<Props> {
  static defaultProps = defaultProps;

  render() {
    const {
      tag,
      hrefString,
      children,
      loading,
      shape,
      size,
      primary,
      secondary,
      icon,
      iconPosition,
      variant,

    } = this.props;

   const ListButton = {
     raised: Btn,
     outline: OutlineBtn,
     inverted: OutlineBtn,
     icon: IconBtn
   }
   const hasChild = !isChild(children);

    const ElementType = !hrefString ? ListButton[variant].withComponent(tag) : ListButton[variant].withComponent('a')

    return (
      <ElementType {...this.props}>
        {
          icon
            ? (
              loading
              ? (<SpinnerButton size={size} iconPosition={iconPosition} shape={shape} primary={primary} secondary={secondary}/>)
              : (<Icon iconPosition={iconPosition} name={icon}/>))
            : ('')
        }
        {loading && !icon && <SpinnerButton size={size} iconPosition={iconPosition} shape={shape} primary={primary} secondary={secondary}/>}
        {
          shape === "circle" || variant === "icon"
            ? (null)
            : (hasChild && <Label>
              {children}
            </Label>)
        }

      </ElementType>
    )
  }
}

export default Button
