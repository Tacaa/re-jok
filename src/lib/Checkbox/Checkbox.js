// @flow
import * as React from 'react';
import {
  StyledCheckbox,
  StyledCheckboxLabel,
  StyledCheckboxWrapper,
  StyledCheckboxButton,
} from './style';

import CheckboxGroup from './CheckboxGroup';

import { withTheme } from 'styled-components';

import { isUndef, isChild } from '../helpers/typeUtils';
import { generalId } from '../helpers';

type Props = {
  /**Override style of Checkbox**/
  style?: Object,
  /** Add more class to Checkbox**/
  className?: string,
  /****/
  children?: any,
  /** onChange callback function... params is (e: Event, props: Object)**/
  onChange?: Function,
  /** Set value of Checkbox**/
  value?: string | number,
  /** Set name of checkbox component**/
  name: string,
  /** Set checkbox component disabled or not**/
  disabled: boolean,
  /** Set checked checkbox property**/
  checked: boolean,
  /** Set checkbox to indeterminate mode.. not work with toggle mode**/
  indeterminate: boolean,
  /** Label of checkbox**/
  label?: string,
  /** Set color of active checkbox and active toggle**/
  color?: string,
  /** Change checkbox to toggle mode**/
  toggle: boolean,
  /**Only work with toggle mode**/
  inactiveColor: string


}

const defaultProps = {
  checked: void 0,
  disabled: false,
  indeterminate: false,
  toggle: false,
}

type State = {
  checked: boolean
}


class Checkbox extends React.Component<Props, State>{
  static defaultProps = defaultProps;

  static Group = CheckboxGroup;

  inputId: string;

  constructor(props: Props){
    super(props);

    this.inputId = generalId();

    this.state = {
      checked: props.checked || false,
    }
  }

  componentWillReceiveProps(nextProps: Props) {
   if (nextProps.checked !== this.props.checked && isUndef(nextProps.checked)) {
     this.setState({
       checked: this.props.checked,
     });
   }
 }


  handleChange = (e: SyntheticEvent<HTMLInputElement>) => {
    const {onChange} = this.props;

    const checked = (e.target: window.HTMLInputElement).checked;

    isUndef(this.props.checked) && this.setState({ checked });

    if(onChange) onChange(e, {...this.props, checked: checked});

  }


  render(){
    const {
      children,
      onChange,
      name,
      disabled,
      indeterminate,
      label,
      color,
      inactiveColor,
      toggle,
      ...rest
    } = this.props;

    const hasChild = !isChild(children);

    const checked = isUndef(this.props.checked) ? this.state.checked : this.props.checked;

    return(
      <StyledCheckboxWrapper>
        <StyledCheckbox
          checked={checked}
          name={name}
          id={this.inputId}
          disabled={disabled}
          onChange={this.handleChange}
          indeterminate={indeterminate}
          {...rest}
         />
      <StyledCheckboxLabel
        id={this.inputId}
        toggle={toggle}
        color={color}
        disabled={disabled}
        indeterminate={indeterminate}

        >
        <StyledCheckboxButton
          toggle={toggle}
          checked={checked}
          indeterminate={indeterminate}
          disabled={disabled}
          inactiveColor={inactiveColor}
          color={color}
        />
        {label ? label : hasChild && children}
      </StyledCheckboxLabel>
      </StyledCheckboxWrapper>
    )
  }
}


export default withTheme(Checkbox)
