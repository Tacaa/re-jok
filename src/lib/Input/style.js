import styled, {css} from 'styled-components';

export const StyledInputWrapper = styled.div`
   box-sizing: border-box;
   margin: 0;
   padding: 0;
   list-style: none;
   position: relative;
   display: table;
   width: ${props => props.fullWidth && "100%"};

`



export const StyledInput = styled.input`

appearance: none;
max-width: 100%;
width: 100%;
height: ${props => props.theme.size[props.size].height};

padding: ${props => props.theme.size[props.size].padding};
font-size: ${props => props.theme.size[props.size].fontSize};

color: ${({theme}) => theme.text.regular};
background-image: none;


box-sizing: border-box;
transition: all .2s ease-in-out;
${
  props =>
    props.blank
    ? css`
    background: none;
    border-color: transparent;
    &:focus{
      outline: 0;
    }
    `
    : css`
    border: ${({theme}) => `.1rem solid ${theme.border.primary}`};
    background: ${({theme}) => `${theme.background.default}`};
    border-radius: ${props => props.addonAfter || props.addonBefore ? "" : ".4rem"};
    border-color: ${({theme, validateStatus, underline}) => validateStatus && !underline && theme[validateStatus].main};
    border-bottom: ${({theme, validateStatus, underline}) => validateStatus && underline && `.2rem solid ${theme[validateStatus].main}`};

    &:hover{

       border-color: ${({theme, validateStatus, underline}) =>validateStatus && !underline ? theme[validateStatus].main : `#c0c4cc`};
       border-bottom: ${({theme, validateStatus, underline}) => validateStatus && underline && `.2rem solid ${theme[validateStatus].main}`};


    }

    &:focus {
        outline: none;
        ${
          props =>
             props.underline ? css`

                 border-bottom: ${({theme, validateStatus,colorFocus}) => `.2rem solid ${validateStatus ? theme[validateStatus].main : colorFocus ? colorFocus : theme.primary.main}`};
                 box-shadow: 0 .5rem 1rem rgba(0,0,0,0.1);

             `: css`
               ${
                 props =>
                  props.validateStatus
                  ? css`
                  border-color: ${({theme, validateStatus}) => theme[validateStatus].main};

                  `
                  : css`
                  border-color: ${({theme, colorFocus}) => colorFocus ? colorFocus : theme.primary.main};

                  `
               }

             `
        }
    }
    `
}


${
  props => props.textarea && css`
    overflow-y: ${props => props.autosize && "hidden"};
    font-family: sans-serif;
    height: auto;
  `
}




${
  props =>
    props.icon && css`
    padding-left: ${props => props.iconPosition === "left" && "3.2rem"};
    padding-right: ${props => props.iconPosition === "right" && "3.2rem"};
 `
}

${
  props =>
    props.disabled && css`
    border-color: #d9d9d9;
    cursor: not-allowed;
    background-color: #f5f5f5;
    `
}




`

export const StyledIconInput = styled.div`
    position: absolute;
    top: 0;
    bottom: 0;
    left: ${props => props.iconPosition === "left" && 0};
    right: ${props => props.iconPosition === "right" && 0};
    width: 3.2rem;
    display: inline-flex;
    justify-content: center;
    align-items: center;
    color: ${({theme}) => theme.text.regular};
    pointer-events: none;
`



const baseAddon = css`
  padding: 0 1.1rem;
  font-size: 1.4rem;
  font-weight: normal;
  line-height: 1;
  color: ${({theme}) => theme.text.regular};
  text-align: center;
  background-color: ${({theme}) => theme.background.default};
  border: ${({theme}) => `.1rem solid ${theme.border.primary}`};
  position: relative;
  transition: all .3s;
  display: table-cell;
  ${
    ({validateStatus, theme}) => validateStatus && css`
      color: ${theme[validateStatus].main};
      border-color: ${theme[validateStatus].main};
      background-color: ${({theme}) => theme.background.primary};
    `
  }
`

export const StyledAddonBefore = styled.div`
  ${baseAddon};
  border-right: 0;
`

export const StyledAddonAfter = styled.div`
  ${baseAddon};
  border-left: 0;
`
