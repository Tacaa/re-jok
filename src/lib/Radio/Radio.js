// @flow
import * as React from 'react';
import {
  StyledRadioWrapper,
  StyledRadio,
  StyledRadioLabel,
  StyledRadioButton
} from './style';

import { withTheme } from 'styled-components';

import RadioGroup from './RadioGroup';
import { isUndef, isChild, isFunction } from '../helpers/typeUtils';
import { generalId, noop } from '../helpers';


type Props = {
  /**Add style of Radio**/
  style?: Object,
  /** Add more class to Radio**/
  className?: string,
  /**Children could be anything**/
  children?: any,
  /** onChange callback function... params is (e: Event, props: Object)**/
  onChange: Function,
  /** Set value of Radio**/
  value?: string | number,
  /** Set name of radio component**/
  name: string,
  /** Set radio component disabled or not**/
  disabled: boolean,
  /** Set checked radio property**/
  checked: boolean,
  /** Set radio default checked or not**/
  defaultChecked: boolean,
  /** Label of radio**/
  label?: string,
  /** Set color of radio**/
  color?: string,

}

const defaultProps = {
  checked: void 0,
  defaultChecked: false,
  disabled: false,
  onChange: noop,
}

type State = {
  checked: boolean
}

class Radio extends React.Component<Props, State> {

  static defaultProps = defaultProps;

  static Group = RadioGroup;

  inputId: string;

  constructor(props: Props){
    super(props);

    this.inputId = generalId();

    this.state = {
      checked: isUndef(props.checked)
      ? props.defaultChecked
      : props.checked,
    }
  }



  handleChange = (e: SyntheticEvent<HTMLInputElement>) => {
    const {onChange} = this.props;

    const checked = (e.target: window.HTMLInputElement).checked;

    isFunction(onChange) && onChange(e, {...this.props});

    isUndef(this.props.checked) && this.setState({ checked });
  }

  render(){
    const {
      children,
      onChange,
      name,
      disabled,
      label,
      color,
      ...rest
    } = this.props;

    const hasChild = !isChild(children);

    const checked = isUndef(this.props.checked) ? this.state.checked : this.props.checked;


    return (
      <StyledRadioWrapper>

        <StyledRadio
          onChange={this.handleChange}
          checked={checked}
          name={name}
          id={this.inputId}
          disabled={disabled}
          value={rest.value}
        />
      <StyledRadioLabel
        id={this.inputId}
        color={color}
        disabled={disabled}
        {...rest}
        >
        <StyledRadioButton
          checked={checked}
          disabled={disabled}
          color={color}

        />

      {label ? label : hasChild && children}
      </StyledRadioLabel>
      </StyledRadioWrapper>
    )
  }
}


export default withTheme(Radio)
