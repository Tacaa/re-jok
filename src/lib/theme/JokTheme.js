// @flow
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from './index';

type Props = {
  /**Theme of project**/
  theme: Object,
  /**Children can be anything**/
  children: any,
}

const defaultProps = {
  theme: {}
}

class JokTheme extends React.Component<Props>{
  static defaultProps= defaultProps;

  checkDefaultTheme = ()=>{
    const {theme} = this.props;
    if(Object.keys(theme).length === 0 && theme.constructor === Object) return true;
    return false;
  }

  render(){
    const {
      children,
      theme
    } = this.props;

    let useDefaultTheme = this.checkDefaultTheme();

    return (
      <ThemeProvider theme={useDefaultTheme? defaultTheme : theme}>
        {children}
      </ThemeProvider>
    )
  }
}

export default JokTheme
