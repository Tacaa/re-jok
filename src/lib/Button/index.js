import Button from './Button';

import TextButton from './TextButton';


export default {
  Button,
  TextButton
};