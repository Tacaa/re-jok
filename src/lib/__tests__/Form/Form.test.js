import React from 'react';
import Form from '../../Form/Form';
import Icon from '../../Icon';
import Input from '../../Input/Input';
import Radio from '../../Radio/Radio';
import Checkbox from '../../Checkbox/Checkbox';
import Button from '../../Button/Button';
import Select from '../../Select/Select';
import { defaultTheme } from '../../theme';

import { shallowWithTheme, mountWithTheme, renderWithTheme } from '../../helpers/withTheme';

import 'jest-enzyme';
import 'jest-styled-components';

//Form
describe('<Form />', () => {
  it('should render Form', () => {
    const tree = renderWithTheme(<Form></Form>).toJSON();
    expect(tree).toMatchSnapshot();
  });



})

//Input
describe('<Input component />', () => {
  it('should render Form and Input component', () => {
    const tree = renderWithTheme(<Form>
      <Form.Item>
        <Input/>
      </Form.Item>
    </Form>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a input size small', () => {
    const wrapper = mountWithTheme(<Input size="small"/>, defaultTheme);
    expect(wrapper.props().size).toEqual('small');
    expect(wrapper.find('input')).toHaveLength(1);

    const tree = renderWithTheme(<Input size="small"/>).toJSON();
    expect(tree).toMatchSnapshot();



  });

  it('renders a input size default', () => {
    const wrapper = mountWithTheme(<Input size="default"/>, defaultTheme);
    expect(wrapper.props().size).toEqual('default');
    expect(wrapper.find('input')).toHaveLength(1);

    const tree = renderWithTheme(<Input size="default"/>).toJSON();
    expect(tree).toMatchSnapshot();

  });

  it('renders a input size medium', () => {
    const wrapper = mountWithTheme(<Input size="medium"/>, defaultTheme);
    expect(wrapper.props().size).toEqual('medium');
    expect(wrapper.find('input')).toHaveLength(1);

    const tree = renderWithTheme(<Input size="medium"/>).toJSON();
    expect(tree).toMatchSnapshot();

  });

  it('renders a input size large', () => {
    const wrapper = mountWithTheme(<Input size="large"/>, defaultTheme);
    expect(wrapper.props().size).toEqual('large');
    expect(wrapper.find('input')).toHaveLength(1);

    const tree = renderWithTheme(<Input size="large"/>).toJSON();
    expect(tree).toMatchSnapshot();

  });

  it('renders a input as a area', () => {
    const wrapper = mountWithTheme(<Input textarea/>, defaultTheme);
    expect(wrapper.props().textarea).toEqual(true);
    expect(wrapper.find('textarea')).toHaveLength(1);

    const tree = renderWithTheme(<Input textarea/>).toJSON();
    expect(tree).toMatchSnapshot();



  });

  it('renders a underline input', () => {
    const wrapper = mountWithTheme(<Input underline/>, defaultTheme);
    expect(wrapper.props().underline).toEqual(true);
    expect(wrapper.find('input')).toHaveLength(1);

    const tree = renderWithTheme(<Input underline/>).toJSON();
    expect(tree).toMatchSnapshot();



  });

  it('renders a disabled input', () => {
    const wrapper = mountWithTheme(<Input disabled/>, defaultTheme);
    expect(wrapper.props().disabled).toEqual(true);
    expect(wrapper.find('input')).toHaveLength(1);

    const tree = renderWithTheme(<Input disabled/>).toJSON();
    expect(tree).toMatchSnapshot();



  });

  it('renders a blank input', () => {
    const wrapper = mountWithTheme(<Input blank/>, defaultTheme);
    expect(wrapper.props().blank).toEqual(true);
    expect(wrapper.find('input')).toHaveLength(1);

    const tree = renderWithTheme(<Input blank/>).toJSON();
    expect(tree).toMatchSnapshot();



  });

  it('renders a input with Icon', () => {
    const wrapper = mountWithTheme(<Input icon="rocket"/>, defaultTheme);
    expect(wrapper.props().icon).toEqual("rocket");
    expect(wrapper.find(Icon)).toHaveLength(1);

    const tree = renderWithTheme(<Input icon="rocket"/>).toJSON();
    expect(tree).toMatchSnapshot();



  });







})

//Radio
// describe('<Radio />', () => {
//   it('should render Radio', () => {
//     const tree = renderWithTheme( <Radio value={1}>A</Radio>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//
//   it('should render Radio Group', () => {
//     const tree = renderWithTheme(<Radio.Group name="q1" >
//       <Radio value={1}>A</Radio>
//       <Radio value={2}>B</Radio>
//       <Radio value={3}>C</Radio>
//       <Radio value={4}>D</Radio>
//   </Radio.Group>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//
//   it('should render Radio Group with color', () => {
//     const tree = renderWithTheme(<Radio.Group  color="#52c41a" name="q1" value={1} >
//       <Radio value={1}>A</Radio>
//       <Radio value={2}>B</Radio>
//       <Radio value={3}>C</Radio>
//       <Radio value={4}>D</Radio>
//   </Radio.Group>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//
//
//
// })


//Checkbox
// describe('<Checkbox />', () => {
//   it('should render Checkbox', () => {
//     const tree = renderWithTheme(<Checkbox checked value="Apple"/>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//
//   it('should render Checkbox Group', () => {
//     const tree = renderWithTheme( <Checkbox.Group name="q2">
//        <Checkbox value="Apple">Apple</Checkbox>
//        <Checkbox value="Pear">Pear</Checkbox>
//        <Checkbox value="Orange">Orange</Checkbox>
//    </Checkbox.Group>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//
//   it('should render Checkbox Group with color', () => {
//     const tree = renderWithTheme( <Checkbox.Group color="#faad14" name="q2">
//        <Checkbox value="Apple">Apple</Checkbox>
//        <Checkbox value="Pear">Pear</Checkbox>
//        <Checkbox value="Orange">Orange</Checkbox>
//
//    </Checkbox.Group>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//
//   it('should render Checkbox with toggle mode', () => {
//     const tree = renderWithTheme(  <Checkbox checked toggle label="Toggle"/>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
//
//
//
//
// })


//Select
describe('<Select />', () => {
  it('should render Select', () => {
    const tree = renderWithTheme(<Select placeholder="Choose option">
      <Select.Option value={1}>Option 1</Select.Option>
      <Select.Option value={2} disabled>Option 2</Select.Option>
      <Select.Option value={3}>Option 3</Select.Option>

    </Select>).toJSON();
    expect(tree).toMatchSnapshot();
  });





})
