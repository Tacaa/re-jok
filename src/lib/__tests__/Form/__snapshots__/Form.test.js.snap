// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render Form 1`] = `
.c0 {
  font-size: 1.4rem;
  position: relative;
  max-width: 100%;
  color: rgba(0,0,0,.64);
  line-height: 1.5;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

<form
  className="c0"
/>
`;

exports[`<Input component /> renders a blank input 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 3.2rem;
  padding: 0.9rem 1.8rem;
  font-size: 1.4rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  background: none;
  border-color: transparent;
}

.c1:focus {
  outline: 0;
}

<div
  className="c0"
>
  <input
    className="c1"
    disabled={false}
    icon={undefined}
    onChange={[Function]}
    onKeyDown={[Function]}
    size="default"
  />
</div>
`;

exports[`<Input component /> renders a disabled input 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 3.2rem;
  padding: 0.9rem 1.8rem;
  font-size: 1.4rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  border: .1rem solid #e8e8e8;
  background: #fff;
  border-radius: .4rem;
  border-color: #d9d9d9;
  cursor: not-allowed;
  background-color: #f5f5f5;
}

.c1:hover {
  border-color: #c0c4cc;
}

.c1:focus {
  outline: none;
  border-color: #1c8ffb;
}

<div
  className="c0"
>
  <input
    className="c1"
    disabled={true}
    icon={undefined}
    onChange={[Function]}
    onKeyDown={[Function]}
    size="default"
  />
</div>
`;

exports[`<Input component /> renders a input as a area 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 3.2rem;
  padding: 0.9rem 1.8rem;
  font-size: 1.4rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  border: .1rem solid #e8e8e8;
  background: #fff;
  border-radius: .4rem;
  font-family: sans-serif;
  height: auto;
}

.c1:hover {
  border-color: #c0c4cc;
}

.c1:focus {
  outline: none;
  border-color: #1c8ffb;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={false}
    icon={undefined}
    onChange={[Function]}
    onKeyDown={[Function]}
    size="default"
  />
</div>
`;

exports[`<Input component /> renders a input size default 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 3.2rem;
  padding: 0.9rem 1.8rem;
  font-size: 1.4rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  border: .1rem solid #e8e8e8;
  background: #fff;
  border-radius: .4rem;
}

.c1:hover {
  border-color: #c0c4cc;
}

.c1:focus {
  outline: none;
  border-color: #1c8ffb;
}

<div
  className="c0"
>
  <input
    className="c1"
    disabled={false}
    icon={undefined}
    onChange={[Function]}
    onKeyDown={[Function]}
    size="default"
  />
</div>
`;

exports[`<Input component /> renders a input size large 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 4.8rem;
  padding: 1.8rem 3.6rem;
  font-size: 2rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  border: .1rem solid #e8e8e8;
  background: #fff;
  border-radius: .4rem;
}

.c1:hover {
  border-color: #c0c4cc;
}

.c1:focus {
  outline: none;
  border-color: #1c8ffb;
}

<div
  className="c0"
>
  <input
    className="c1"
    disabled={false}
    icon={undefined}
    onChange={[Function]}
    onKeyDown={[Function]}
    size="large"
  />
</div>
`;

exports[`<Input component /> renders a input size medium 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 4rem;
  padding: 1.4rem 2.7rem;
  font-size: 1.8rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  border: .1rem solid #e8e8e8;
  background: #fff;
  border-radius: .4rem;
}

.c1:hover {
  border-color: #c0c4cc;
}

.c1:focus {
  outline: none;
  border-color: #1c8ffb;
}

<div
  className="c0"
>
  <input
    className="c1"
    disabled={false}
    icon={undefined}
    onChange={[Function]}
    onKeyDown={[Function]}
    size="medium"
  />
</div>
`;

exports[`<Input component /> renders a input size small 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 2.4rem;
  padding: 0.5rem 0.9rem;
  font-size: 1.2rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  border: .1rem solid #e8e8e8;
  background: #fff;
  border-radius: .4rem;
}

.c1:hover {
  border-color: #c0c4cc;
}

.c1:focus {
  outline: none;
  border-color: #1c8ffb;
}

<div
  className="c0"
>
  <input
    className="c1"
    disabled={false}
    icon={undefined}
    onChange={[Function]}
    onKeyDown={[Function]}
    size="small"
  />
</div>
`;

exports[`<Input component /> renders a input with Icon 1`] = `
.c2 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  display: inline-block;
}

.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 3.2rem;
  padding: 0.9rem 1.8rem;
  font-size: 1.4rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  border: .1rem solid #e8e8e8;
  background: #fff;
  border-radius: .4rem;
  padding-left: 3.2rem;
}

.c3:hover {
  border-color: #c0c4cc;
}

.c3:focus {
  outline: none;
  border-color: #1c8ffb;
}

.c1 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  width: 3.2rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(0,0,0,.64);
  pointer-events: none;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      color={undefined}
      style={
        Object {
          "fontSize": "1.4rem",
        }
      }
    >
      <span
        aria-hidden={true}
        className="fa fa-rocket"
      />
    </div>
  </div>
  <input
    className="c3"
    disabled={false}
    icon="rocket"
    onChange={[Function]}
    onKeyDown={[Function]}
    size="default"
  />
</div>
`;

exports[`<Input component /> renders a underline input 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 3.2rem;
  padding: 0.9rem 1.8rem;
  font-size: 1.4rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  border: .1rem solid #e8e8e8;
  background: #fff;
  border-radius: .4rem;
}

.c1:hover {
  border-color: #c0c4cc;
}

.c1:focus {
  outline: none;
  border-bottom: .2rem solid #1c8ffb;
  box-shadow: 0 .5rem 1rem rgba(0,0,0,0.1);
}

<div
  className="c0"
>
  <input
    className="c1"
    disabled={false}
    icon={undefined}
    onChange={[Function]}
    onKeyDown={[Function]}
    size="default"
  />
</div>
`;

exports[`<Input component /> should render Form and Input component 1`] = `
.c0 {
  font-size: 1.4rem;
  position: relative;
  max-width: 100%;
  color: rgba(0,0,0,.64);
  line-height: 1.5;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.c1 {
  font-size: 1.4rem;
  position: relative;
  max-width: 100%;
  color: rgba(0,0,0,.64);
  line-height: 1.5;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  margin-bottom: 1.4rem;
  clear: both;
  vertical-align: top;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  position: relative;
  display: table;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  max-width: 100%;
  width: 100%;
  height: 3.2rem;
  padding: 0.9rem 1.8rem;
  font-size: 1.4rem;
  color: rgba(0,0,0,.64);
  background-image: none;
  box-sizing: border-box;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  border: .1rem solid #e8e8e8;
  background: #fff;
  border-radius: .4rem;
}

.c3:hover {
  border-color: #c0c4cc;
}

.c3:focus {
  outline: none;
  border-color: #1c8ffb;
}

<form
  className="c0"
>
  <div
    className="c1"
    required={false}
  >
    <div
      className=""
    >
      <div
        className="c2"
      >
        <input
          className="c3"
          disabled={false}
          icon={undefined}
          onChange={[Function]}
          onKeyDown={[Function]}
          required={false}
          size="default"
        />
      </div>
    </div>
  </div>
</form>
`;

exports[`<Select /> should render Select 1`] = `
.c0 {
  max-width: 100%;
  width: 100%;
  border: 0 none;
  padding: 0 1rem;
  background: #fff;
  color: rgba(0,0,0,.64);
  border: 1px solid #e8e8e8;
  -webkit-transition: .2s ease-in-out;
  transition: .2s ease-in-out;
  -webkit-transition-property: color,background-color,border;
  transition-property: color,background-color,border;
  height: 3.2rem;
  vertical-align: middle;
  display: inline-block;
  cursor: pointer;
  border-radius: .2rem;
}

.c0:hover {
  border-color: #c0c4cc;
}

.c0:focus {
  outline: none;
}

<select
  className="c0"
  onChange={[Function]}
>
  <option
    className=""
    disabled={true}
    value={-1}
  >
    Choose option
  </option>
  <option
    className=""
    disabled={false}
    value={1}
  >
    Option 1
  </option>
  <option
    className=""
    disabled={true}
    value={2}
  >
    Option 2
  </option>
  <option
    className=""
    disabled={false}
    value={3}
  >
    Option 3
  </option>
</select>
`;
